Allphi/Domain/Domaincontroller.cs

```diff
@@ -73,17 +74,127 @@ public ParkingSpot GetAvailableParkingSpotVisitor()
             return (_parkingRepo.GetAvailableParkingSpotUnreserved());
         }
 
-        #endregion Parking
-
-        public void SubmitVisitor(string licensePlate)
+        public void SubmitVisitorParking(string licensePlate)
         {
             string license = Guard.Against.NullOrEmpty(licensePlate, nameof(licensePlate));
-            ParkingSpot parkingSpot = GetAvailableParkingSpotVisitor();
+            ParkingSpot parkingSpot = _parkingRepo.GetAvailableParkingSpotUnreserved();
             if (IsLicensePlateValid(license) && parkingSpot != null && _parkingRepo.CountParkingSpotByPlate(license) == 0)
             {
                 parkingSpot.Plate = license;
                 _parkingRepo.UpdateParkingSpot(parkingSpot);
             };
         }
+
+        public void SubmitEmployeeParking(string licensePlate)
```

Comment:
Consider moving part of this logic to the inner classes rather than the domain controller itself, you might end up creating a monolith otherwise.